sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -111,23 +130,38 @@
             ConfidentialClientApplication.Builder applicationBuilder =
                 ConfidentialClientApplication.builder(clientId, ClientCredentialFactory.createFromSecret(clientSecret))
                     .authority(authorityUrl);
-            if (options.getProxyOptions() != null) {
-                applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
-            }
-            if (options.getHttpPipeline() != null) {
+
+            // If http pipeline is available, then it should override the proxy options if any configured.
+            if (httpPipeline != null) {
                 applicationBuilder.httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));
```

Comment:
I believe this should be `applicationBuilder.httpClient(new HttpPipelineAdapter(httpPipeline));` rather than `applicationBuilder.httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));` because `httpPipeline` could have been set directly from the options or constructed in the ctor, so I think it's possible that `options.getHttpPipeline()` is null when `httpPipeline` is not.
 
Also, the client creates new `HttpPipelineAdapter` instances for each msal client it creates. Does each msal client need it's own adapter? If not I would suggest caching the `HttpPipelineAdapter` rather than just the `HttpPipeline`

Reply:
added the logic to cache the pipeline adapter.
Thanks for the catch on http pipeline, it was an oversight by me.